pystiche/papers/sanakoyeu_et_al_2018/core.py

```diff
@@ -0,0 +1,195 @@
+from typing import Optional, Callable, Union
+import itertools
+import torch
+from torch.utils.data import DataLoader
+from torch import nn
+from torch.optim.optimizer import Optimizer
+from .modules import (
+    SanakoyeuEtAl2018Generator,
+    SanakoyeuEtAl2018Discriminator,
+    sanakoyeu_et_al_2018_discriminator,
+    SanakoyeuEtAl2018TransformerBlock,
+)
+from .loss import (
+    SanakoyeuEtAl2018GeneratorLoss,
+    DiscriminatorLoss,
+    sanakoyeu_et_al_2018_generator_loss,
+)
+from .data import (
+    sanakoyeu_et_al_2018_dataset,
+    sanakoyeu_et_al_2018_image_loader,
+    sanakoyeu_et_al_2018_images,
+)
+from .utils import sanakoyeu_et_al_2018_optimizer
+
+__all__ = [
+    "sanakoyeu_et_al_2018_training",
+    "sanakoyeu_et_al_2018_stylization",
+    "SanakoyeuEtAl2018Generator",
+    "sanakoyeu_et_al_2018_discriminator",
+    "SanakoyeuEtAl2018TransformerBlock",
+    "sanakoyeu_et_al_2018_generator_loss",
+    "DiscriminatorLoss",
+    "sanakoyeu_et_al_2018_images",
+    "sanakoyeu_et_al_2018_image_loader",
+    "sanakoyeu_et_al_2018_dataset"
+]
+
+
+def default_gan_optim_loop(
+    content_image_loader: DataLoader,
+    style_image_loader: DataLoader,
+    generator: nn.Module,
+    discriminator: nn.Module,
+    transformer_block: nn.Module,
```

Comment:
Because the `transformer_block` is only a weak image transformation used by the author to calculate the transformed image loss. So this block is not a `transformer` that creates a `stylized_image` from an `input_image`, like in the papers before. But you are right, the name is misleading in relation to the other papers and should be renamed. Maybe something like `weak_image_transformation`, as the author also called it in the paper. Or what do you think? 

The current name is based on the name in the figure of the author's model.

Reply:
In that case I understood something wrong in the paper. Let me read that again and get back to you.

Reply:
So `generator` here is what we usually called `transformer`, right? What is `transformer_block` then? The phrase "transformer block" is used in the paper for the partial content loss. As such that should be part of the `generator_criterion`.

Reply:
Right the `generator` is what we usually called `transformer`. I will rename it according to the other papers. 
You're right. The `transformer_block` should be handled in the `generator_criterion` and not there.